micro_sam/sam_annotator/image_series_annotator.py

```diff
@@ -156,11 +192,10 @@ def next_image(*args):
         if state.amg is not None:
             state.amg.clear_state()
         state.initialize_predictor(
-            image, model_type=model_type, ndim=2, save_path=image_embedding_path,
-            halo=halo, tile_shape=tile_shape, predictor=predictor,
-            precompute_amg_state=precompute_amg_state, device=device,
+            image, model_type=model_type, ndim=3 if is_volumetric else 2, save_path=image_embedding_path, halo=halo,
+            tile_shape=tile_shape, predictor=predictor, precompute_amg_state=precompute_amg_state, device=device,
         )
-        state.image_shape = image.shape[:-1] if image.ndim == 3 else image.shape
+        state.image_shape = image.shape
```

Comment:
You need to use the same logic as above. Use refactoring into a helper function (see comment above).